### PR TITLE
Gitlab auth issue

### PR DESCRIPTION
**Pull request by lyda**
This fixes gitlab.

I also added a way to debug issues with gitlab authentication in the future. Maybe we can do something like this for the other providers?

### CODE REVIEW

File: server/auth/gitlab/gitlab.go [hunk 1]
```diff
@@ -19,14 +24,98 @@ const (
 // Gitlab account.
 type Config struct {
 	config    *oauth2.Config
-	baseurl   string
 	group     string
 	whitelist map[string]bool
 	allusers  bool
+	apiurl    string
```
Comment (nsheridan) [inline server/auth/gitlab/gitlab.go]: The README needs to be updated with these changes.

File: server/auth/gitlab/gitlab.go [hunk 2]
```diff
@@ -19,14 +24,98 @@ const (
...
+	url := fmt.Sprintf("%sgroups/%s/members/%d", c.apiurl, group, uid)
+	resp, err := client.Get(url)
+	if err != nil {
+		c.logMsg(fmt.Sprintf("Failed to get groups: %s", err))
+		return false
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		if c.log {
+			var body bytes.Buffer
```
Comment (nsheridan) [inline server/auth/gitlab/gitlab.go]: This is duplicated code from above. Maybe it should be a function?

File: server/handlers.go [hunk 3]
```diff
@@ -112,7 +115,9 @@ func (a *app) auth(w http.ResponseWriter, r *http.Request) {
 }
 
 func (a *app) index(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Entering index handler.")
 	tok := a.getAuthToken(r)
+	log.Printf("Token found: %v\n", tok)
```
Comment (nsheridan) [inline server/handlers.go]: I don't think the token should be logged.

Comment (lyda) [reply server/handlers.go]: Oops! No, sorry, that was a debug thing.  gone.

File: server/auth/gitlab/gitlab.go (hunk 2 above)
Comment (lyda) [reply server/auth/gitlab/gitlab.go]: Done.

File: server/auth/gitlab/gitlab.go (hunk 1 above)
Comment (lyda) [reply server/auth/gitlab/gitlab.go]: Done

**Pull request merged**